src/System.Private.CoreLib/src/System/Threading/Thread.CoreRT.cs

```diff
@@ -34,6 +34,13 @@ public sealed partial class Thread
         // Protects starting the thread and setting its priority
         private Lock _lock;
 
+        // so far the only place we initialize it is `WaitForForegroundThreads`
+        // and only in the case when there are running foreground threads 
+        // by the moment of `StartupCodeHelpers.Shutdown()` invocaiton
+        private static ManualResetEvent _allDone;
+
+        // main thread started by runtime and we have no chance (?) to increment it elsewhere
```

Comment:
A good place to increment this for the main thread is inside `InitializeExistingThread` around the place where we set the background bit.